### DISCUSSION TITLE
Future CC2652P2 Matter/Thread update?

### DISCUSSION THREAD

**Discussion by Drizzt321** [tube0013/tube_gateways#117] [General]
I have one of your older CC2652P2 based ethernet coordinators, with firmware/model `CC1352/CC2652, Z-Stack 3.30+ (build 20210120)`, one of the circular based ones. Given that Matter/Thread is starting to gain momentum, I'm looking ahead at how I'll need to integrate it. I do see you have a new [EFR32 MGM21 PoE](https://www.tubeszb.com/product/efr32mgm21_poe/54) ethernet one that says it will have a concurrent Thread/Matter + ZigBee firmware upgrade in the future. Just wanted to confirm, this is what I'll end up needing for the future? And will my existing CC2652P2 routers I also purchased from you be able to be used for Thread/Matter? Or will you release additional Thread/Matter + ZigBee router in the future?

**Comment by tube0013** [comment]
No current roadmap from TI that I've seen that will enable concurrent zigbee and thread. It's theoretically possible that the solution by silicon labs could be used on other radios but still require specific fw features for tagging the incoming data so it can be directed by a daemon to zigbee or openthread border router.

It may not be a bad idea to just get another device to run the thread for matter. The standalone zigbee routers will stay zigbee only. The module is the same as used in coordinators so with additional hardware you could use it to create a border router for thread. 🤔

It should be possible to use an esp32 and so this. But it would be kind of technical.

**Reply by Hedda** [reply]
> It should be possible to use an esp32 and so this. But it would be kind of technical.

FYI, **ESP32-C6** SoC, module and development board were very recently launched has WiFi 6, BLE 5.0, and 802.15.4, including OpenThread and Zigbee (though I understand that Zigbee and Thread are currently exclusive so have to choose one or the other right now).

Maybe that could work as a replacement ff the ESP32-C3 in order to make a device with both Thread Board Router and Zigbee Coordinator?

https://www.espressif.com.cn/en/news/ESP32-C6_Available

https://espressif-docs.readthedocs-hosted.com/projects/espressif-esp-dev-kits/en/latest/esp32c6/esp32-c6-devkitc-1/index.html

https://github.com/espressif/esp-idf/issues/10423

Espressif launched the ESP32-C6 WiFi 6, Bluetooth 5.0 LE, and 802.15.4 IoT microcontroller by making the ESP32-C6-WROOM-1 module:

https://www.cnx-software.com/2021/04/11/esp32-c6-wifi-6-and-bluetooth-le-risc-v-soc-for-iot-devices/

![image](https://user-images.githubusercontent.com/6320001/215800489-abfa75a2-e655-4fe7-8bc9-b5f1850d5cd6.png)

![image](https://user-images.githubusercontent.com/6320001/215800565-c76c8c78-d759-4bd9-833f-1e11e1a36c56.png)

_Espressif Systems has finally launched the ESP32-C6 WiFi 6, Bluetooth 5.0 LE, and 802.15.4 IoT microcontroller by making the ESP32-C6-WROOM-1 module available on [Aliexpress for $14.25](https://fave.co/3VYE3RW) for a pack of 5 modules as well as the ESP32-C6-DevKitC-1 development board that’s [out of stock on the official store](https://fave.co/3vU14ea) at the time of writing. but somehow listed in another store [for $16.22 plus shipping](https://s.click.aliexpress.com/e/_DlvBgFz)._

_Besides the availability announcement, another surprise is the 802.15.4 radio for Zigbee and Thread that was not part of the original [ESP32-C6 announcement](https://www.cnx-software.com/2021/04/11/esp32-c6-wifi-6-and-bluetooth-le-risc-v-soc-for-iot-devices/) in April 2021. The ESP32-C6-DevKitC-1 development board comes with a module with 8MB flash, exposes most of the I/O’s from the ESP32-C3, and comes with two USB-C ports, two buttons, and an RGB LED._

ESP32-C6-DevKitC-1 development board specifications:

- Wireless module  – ESP32-C6-WROOM-1 with
  - ESP32-C6 single-core 32-bit RISC-V clocked up to 160 MHz 320KB ROM, 512KB SRAM, low-power RISC-V core @ up to 20 MHz
  - Wireless – 2.4 GHz WiFi 6, Bluetooth 5.0 LE, 802.15.4 radio for Zigbee 3.0 and Thread. Matter compatible. PCB antenna
8MB SPI flash
- USB – 1x USB Type-C host port, 1x USB Type-C port via UART console
- I/Os – 2x 16-pin headers with GPIO, UART, SPI, ADC, SDIO, USB, I2S, etc…
- Misc – Reset and BOOT buttons, RGB LED, J5 header/jumper for current measurement
- Power Supply (mutually exclusive options)
  - 5V via USB-UART port (recommended)
  - 5V and GND pin headers
  - 3V3 and GND pin headers
- Dimensions – 51.8 x ~20 mm

![image](https://user-images.githubusercontent.com/6320001/215801047-b1044ba0-b1a5-49f5-86f9-de61167a0595.png)

![image](https://user-images.githubusercontent.com/6320001/215801103-6e072b85-e22c-4f30-8d0d-74edb427300a.png)

_The new ESP32-C6 module and development board work with the ESP-IDF v5.1, currently under development (see [progress report](https://github.com/espressif/esp-idf/issues/10423)), and some [basic documentation](https://espressif-docs.readthedocs-hosted.com/projects/espressif-esp-dev-kits/en/latest/esp32c6/esp32-c6-devkitc-1/index.html) can be found on read the docs._

_If you can’t purchase an ESP32-C6 module or board just yet due to limited supplies, don’t despair as more are coming very soon as [Espressif tells](https://www.espressif.com.cn/en/news/ESP32-C6_Available?position=0&list=WQQQ-4A66v8oesBoFKp-rNz9bSQ9dg5w7wc_nNvDt_I) us to expect more products on Adafruit, Akizuki, Digikey, and Mouser._

**Reply by Hedda** [reply]
ESP32-H2 S could also be used as an alternative to the ESP32-C6 SoC if combine the ESP32-H2 SoC with [a chip for LAN Ethernet PHY in order to bridge it to wired LAN connection (using an RJ-45 port with or without PoE Power-Over-Ethernet)](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/hw-reference/esp32/get-started-ethernet-kit.html), or alternatively use an ESP32-H2 SoC in combination with an ESP32-C3 SoC, with the ESP32-H2 SoC only being used as an Thread Border Router (OpenThread Border Router / OTBR).

FYI, before the ESP32-C6 SoC were released this combination of ESP32-H2 + ESP32-C3 was Espressif reference design for a bridge product.

When ESP32-H2 was first released Espressif posted a lot of news on its Espressif's ESP Matter Solution (e.i. "**_esp-matter_**") which uses ESP32 as a Matter-bridge for non-Matter devices such as Zigbee, Thread, and Bluetooth devices, as well as the related news about ESP32-H2 SoC (and "ESP32-H2 DevKitC" board with "ESP32-H2-WROOM-1" radio module) with more news from Espressif about newly established support for Thread/OpenThread (e.i. "**_esp-thread_**") + ZBOSS Open Initiative (ZOI) and Zigbee certification announcements (e.i. "**_esp-zigbee_**").

In essence, it seems like Espressif was focusing on offering a full design for a Zigbee gateway and Zigbee/Thread to Matter bridge implementation (with an all-in-one two-SoC gateway/bridge solution combining an ESP32-H2 for Zigbee/Thread with an ESP32-S3 for Wi-Fi). For that Zigbee/Thread/BLE to Matter-bridge solution they now even have two reference hardware designs that I linked below. 

"**_We offer both Matter-Zigbee and Matter-BLE Mesh bridge solutions with full functional software SDK support. A Matter-Zigbee Bridge uses two SoCs (Wi-Fi + 802.15.4), they are connected via a serial interface like UART or SPI, while a Matter-BLE Mesh Bridge can be done on single SoC with both Wi-Fi and BLE interfaces._**"

![image](https://user-images.githubusercontent.com/6320001/193288233-f01d99aa-031d-4e12-902c-5bfa36dfaf7d.png)

![image](https://user-images.githubusercontent.com/6320001/193288824-3d642559-6b1a-4888-8164-b4a3a2230a69.png)

### Espressif ESP Matter news:

https://www.espressif.com/en/news/ESP_Matter_Solution

https://www.espressif.com/en/solutions/device-connectivity/esp-matter-solution

https://www.espressif.com/en/news/ESP32_H2

https://www.espressif.com/en/news/ESP32-H2_Thread_and_Zigbee_certified

https://en.dsr-corporation.com/news/espressif-joins-zoi

### esp-matter (Matter / Project CHIP) solutions:

https://medium.com/the-esp-journal/matter-bridge-for-non-matter-devices-d3b7f003a004

**https://blog.espressif.com/matter-thread-border-router-in-matter-240838dc4779**

https://docs.espressif.com/projects/esp-matter/en/main/esp32/

https://github.com/espressif/esp-matter

https://github.com/espressif/esp-matter/tree/main/examples/zigbee_bridge

https://github.com/project-chip/connectedhomeip/tree/master/examples/lighting-app/esp32

https://github.com/project-chip/connectedhomeip/tree/master/examples/chip-tool

### esp-thread:

https://openthread.io/guides/border-router/espressif-esp32

https://github.com/espressif/esp-idf/tree/master/examples/openthread/ot_rcp

https://github.com/espressif/esp-idf/tree/master/examples/openthread/ot_cli

https://github.com/espressif/esp-thread-lib

https://github.com/espressif/esp-thread-br

https://openthread.io/vendors/espressif

### esp-zigbee:

https://github.com/espressif/esp-zigbee-sdk

https://github.com/espressif/esp-zigbee-sdk/tree/main/components/esp-zigbee-lib

https://github.com/espressif/esp-zboss-lib/

https://components.espressif.com/components/espressif/esp-zboss-lib

https://github.com/espressif/esp-idf/tree/master/examples/zigbee/

https://github.com/espressif/esp-idf/blob/master/examples/zigbee/esp_zigbee_rcp

https://github.com/espressif/esp-idf/tree/master/examples/zigbee/esp_zigbee_gateway

https://github.com/openthread/ot-br-posix/

https://openthread.io/platforms/co-processor

### esp-protocols:

esp-protocols including esp modem (for AT commands, etc.), esp websocket client, mDNS (Zeroconf and DNS-SD), Asio, and more:

https://espressif.github.io/esp-protocols/

https://github.com/espressif/esp-protocols

https://github.com/espressif/esp-protocols/tree/master/components/esp_modem

https://github.com/espressif/esp-protocols/tree/master/components/esp_websocket_client

https://github.com/espressif/esp-protocols/tree/master/components/mdns

https://github.com/espressif/esp-protocols/tree/master/components/asio

### Other indirectly related tools from Espressif:

https://github.com/espressif/esptool

https://github.com/espressif/openocd-esp32

https://github.com/espressif/esp-usb-bridge


### Hardware Platforms

"ESP32-H2 DevKitC" ("ESP32-H2-DevKitC-1 V2.1") board with "ESP32-H2-WROOM-1" radio module:

![image](https://user-images.githubusercontent.com/6320001/206166691-081ad290-ec7b-4cdd-bb37-a0b3df675993.png)

The ESP Thread Border Router consists of two SoCs:

* An ESP32 series Wi-Fi SoC (ESP32, ESP32-C, ESP32-S, etc) loaded with ESP Thread Border Router and OpenThread Stack.
* An ESP32-H2 (also a "ESP32-H2-WROOM-1" radio module) with ESP32-H2 802.15.4 SoC loaded with OpenThread RCP.

https://github.com/espressif/esp-thread-br#hardware-platforms

https://github.com/espressif/esp-idf/tree/master/examples/openthread/ot_br

https://openthread.io/guides/border-router/espressif-esp32

#### ESP Thread Border Router Board

The ESP Thread border router board provides an integrated module of an ESP32-S3 SoC and an ESP32-H2 RCP.

![image](https://user-images.githubusercontent.com/6320001/193261039-97f9e77b-9e25-4b4c-a60f-b1584dc2099d.png)

The two SoCs are connected with the following interfaces:
* UART and SPI for serial communication
* RESET and BOOT pins for RCP Update
* 3-Wires PTA for RF coexistence

#### Standalone Modules

The SDK also supports manually connecting an ESP32-H2 RCP to an ESP32 series Wi-Fi SoC.

ESP32 pin | ESP32-H2 pin
----------|-------------
  GND     |      G
  GPIO17  |      TX
  GPIO18  |      RX
  GPIO4   |      RST
  GPIO5   |  GPIO9 (BOOT)

The following image shows an example connection betwe    Add a package diagram. ES32-H2 and ESP32:

![image](https://user-images.githubusercontent.com/6320001/193261559-b90749ee-0298-439b-a366-eb13f6106851.png)

In this setup, only UART interface is connected, so it doesn't support RCP Update or RF Coexistence features. You can refer to [ot_br](https://github.com/espressif/esp-idf/tree/master/examples/openthread/ot_br) example in esp-idf as a quick start.

https://github.com/espressif/esp-idf/tree/master/examples/openthread/ot_br

https://openthread.io/guides/border-router/espressif-esp32

#### Provided Features

These features are currently provided by the SDK:

* **Bi-directional IPv6 Connectivity**: The devices on the backbone link (typically Wi-Fi) and the Thread network can reach each other.
* **Service Discovery Delegate**: The nodes on the Thread network can find the mDNS services on the backbone link.
* **Service Registration Server**: The nodes on the Thread network can register services to the border router for devices on the backbone link to discover.
* **Multicast Forwarding**: The devices joining the same multicast group on the backbone link and the Thread network can be reached with one single multicast.
* **NAT64**: The devices can access the IPv4 Internet via the border router.
* **RCP Update**: The built border router image will contain an updatable RCP image and can automatically update the RCP on version mismatch or RCP failure.
* **Web GUI**: The border router will enable a web server and provide some practical functions including Thread network discovery, network formation and status query. 

In the future releases, the following features will be added:

* SPI Communication
* RF Coexistence
